pkg/virt-controller/services/multus_annotations_test.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package services
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+)
+
+var _ = Describe("Multus annotations", func() {
+	var multusAnnotationPool multusNetworkAnnotationPool
+	var vmi v1.VirtualMachineInstance
+	var network v1.Network
+
+	Context("a multus annotation pool with no elements", func() {
+		BeforeEach(func() {
+			multusAnnotationPool = multusNetworkAnnotationPool{}
+			// for checking the multus annotations, all we need is the namespace
+			vmi = v1.VirtualMachineInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "testvmi", Namespace: "default", UID: "1234",
+				},
+			}
+			network = v1.Network{
+				NetworkSource: v1.NetworkSource{
+					Multus: &v1.MultusNetwork{NetworkName: "test1"},
+				},
+			}
+		})
+
+		It("is empty", func() {
+			Expect(multusAnnotationPool.isEmpty()).To(BeTrue())
+		})
+
+		It("when added an element, is no longer empty", func() {
+			podIfaceName := "net1"
+			multusAnnotationPool.add(newMultusAnnotationData(&vmi, network, podIfaceName))
+			Expect(multusAnnotationPool.isEmpty()).To(BeFalse())
+		})
+
+		It("generate a null string", func() {
+			multusAnnotation, err := multusAnnotationPool.toString()
+			Expect(multusAnnotation).To(BeIdenticalTo("null"))
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("a multus annotation pool with elements", func() {
+		BeforeEach(func() {
+			// for checking the multus annotations, all we need is the namespace
+			vmi = v1.VirtualMachineInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "testvmi", Namespace: "namespace1", UID: "1234",
+				},
+			}
+			network = v1.Network{
+				NetworkSource: v1.NetworkSource{
+					Multus: &v1.MultusNetwork{NetworkName: "test1"},
+				},
+			}
+			multusAnnotationPool = multusNetworkAnnotationPool{
+				pool: []multusNetworkAnnotation{
+					newMultusAnnotationData(&vmi, network, "net1"),
+				},
+			}
+		})
+
+		It("is not empty", func() {
+			Expect(multusAnnotationPool.isEmpty()).To(BeFalse())
+		})
```

Comment:
I think that the check should show that a pool creation is successful. The empty check can be part of it.

Reply:
What would you test there ? Pool creation gives you an empty pool: which does not fit this context.

It's unclear to me what you want.

Reply:
I'm trying to say that the check that it is empty is odd as a test.
Creating an empty pool and checking it is empty, sounds good.
Creating a non empty pool and checking it is not empty, sounds good.

If you really think this makes sense, I won't insist.

Reply:
> "
> Creating an empty pool and checking it is empty, sounds good.

I'm doing this.

 I'm severely confused. In the line above you say -and I quote: "I'm trying to say that the check that it is empty is odd as a test." and now you say creating an empty pool and checking it is empty is good. You can't go left and right at the same time; which one is it ?

> Creating a non empty pool and checking it is not empty, sounds good.

I'm doing this.

> 
> If you really think this makes sense, I won't insist.

OK. Let's leave it at that then.